src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -382,7 +384,14 @@ object RuleUtils {
             baseRelation.schema.contains(s) || (filesDeleted.nonEmpty && s.name.equals(
               IndexConstants.DATA_FILE_NAME_ID))))
 
-        val newLocation = new InMemoryFileIndex(spark, filesToRead, Map(), None)
+        def fileIndex: InMemoryFileIndex =
+          new InMemoryFileIndex(spark, filesToRead, Map(), None)
+        val newLocation = if (filesToRead.length == index.content.files.size) {
+          index.withCachedTag(IndexLogEntryTags.INMEMORYFILEINDEX_INDEX_ONLY)(fileIndex)
+        } else {
+          index.withCachedTag(plan, IndexLogEntryTags.INMEMORYFILEINDEX_HYBRID_SCAN)(fileIndex)
+        }
```

Comment:
I was thinking the following to make the intention clear, but the current approach is also fine:
```scala
        val tagForFileIndex = if (filesToRead.length == index.content.files.size) {
          IndexLogEntryTags.INMEMORYFILEINDEX_INDEX_ONLY
        } else {
          IndexLogEntryTags.INMEMORYFILEINDEX_HYBRID_SCAN
        }
        val newLocation = index.withCachedTag(plan, tagForFileIndex) {
          new InMemoryFileIndex(spark, filesToRead, Map(), None)
        }
```

Reply:
it's because INDEX_ONLY doesn't require plan

Reply:
missed that. thanks